ImportExport/ObjectsToPython/ObjectsToPython.FCMacro

```diff
@@ -0,0 +1,344 @@
+                   ('App::Plane', 'XY_Plane'), ('App::Plane', 'XZ_Plane'), ('App::Plane', 'YZ_Plane'),
+                   ('App::Origin', 'Origin')] 
+    
+    selection = FreeCADGui.Selection.getSelection() 
+    if not selection:
+        selection = doc.Objects
+    else:
+        expandSelection(selection)
+        
+    for obj in doc.Objects:  
+        objectlist.append(doc.getObject(obj.Name)) 
+         
+    for obj in selection:    
+        if obj.TypeId == "Sketcher::SketchObject": 
+            addScriptFunc("")
+            addScriptFunc("def createSketch_%s(doc):"%varname(obj))
+            addObject(doc, obj, objectlist)
+            addScript("return " + varname(obj))
+        
+    addScriptFunc("")
+    addScriptFunc("")
+    addScriptFunc("def make_%s():"%doc.Name)
+    if not selection:
+        addScript("doc = App.newDocument('%s')"%doc.Name)
+    else:
+        addScript("doc = App.ActiveDocument")
+    
+    for obj in selection: 
+        if (obj.TypeId, obj.Label) not in skipObjects:
+            addScriptFunc("")
+            if obj.TypeId == "Sketcher::SketchObject":
+                addScript(varname(obj) + " = createSketch_%s(doc)"%varname(obj))
+            else:
+                addObject(doc, obj, objectlist)            
+                
+    for obj in selection: 
+        if obj.TypeId == 'PartDesign::Body':
+            addScriptFunc("")
+            addScript("%s.Group = %s_Group"%(varname(obj), varname(obj))) 
+            if obj.Tip:
+                addScript("%s.Tip = %s"%(varname(obj), varname(obj.Tip))) 
+            
+    addScript("") 
+    addScript("doc.recompute()") 
+    addScriptFunc("")
+    addScriptFunc("")
+    addScriptFunc("make_%s()"%doc.Name)
+        
+  
+def varname(obj):
+    return obj.Label.replace(" ", "_")
+      
+def addScript(line):
+    FreeCAD.Console.PrintMessage("\t" + line + "\n")
+    
+def addScriptFunc(line):
+    FreeCAD.Console.PrintMessage(line + "\n")
+    
+def expandSelection(selection):    
+    addthis = []     
+    for obj in selection:    
+        if hasattr(obj, 'Group'):
+            addthis += obj.Group
+        if hasattr(obj, 'Source'):
+            addthis.append(obj.Source)
+        if hasattr(obj, 'Links'):
+            addthis += obj.Links
+            
+    added = 0
+    for a in addthis:
+        if (not a in selection) and hasattr(obj, 'Name') and hasattr(obj, 'TypeId'):
+            selection.insert(0, a)
+            added = added + 1
+            
+    if added > 0:
+        expandSelection(selection)
+            
+
+def addObject(doc, obj, objectlist): 
+    objname = varname(obj)
+    addScript("%s = doc.addObject('%s', '%s')"%(objname, obj.TypeId, objname))
+    defaultobj = doc.addObject(obj.TypeId, obj.Name + "Default")
+    addProperties(obj, objname, defaultobj, objectlist)
+    addProperties(obj.ViewObject, objname + ".ViewObject", defaultobj.ViewObject, [])                      
+    doc.removeObject(obj.Name + "Default")      
```

Comment:
I know that there is the possibility to create a hidden temporary document but don't know yet how to use it. Maybe you could consider it and i could use your macro as example.